treescript/src/treescript/github/versionmanip.py

```diff
@@ -36,69 +34,56 @@ def _find_what_version_parser_to_use(file_, source_repo):
     if number_of_version_classes > 0:
         return version_classes[0]
 
-    log.info("Could not determine version class based on file path. Falling back to source_repo")
+    log.info("Could not determine version class based on file path. Falling back to repo")
 
-    version_classes = [cls for repo_name, cls in _VERSION_CLASS_PER_END_OF_SOURCE_REPO.items() if source_repo.endswith(repo_name)]
+    version_classes = [cls for repo_name, cls in _VERSION_CLASS_PER_END_OF_SOURCE_REPO.items() if repo.endswith(repo_name)]
     try:
         return version_classes[0]
     except IndexError as exc:
         raise TreeScriptError(exc) from exc
 
 
-def get_version(file_, parent_directory, source_repo):
-    """Parse the version from file.
+def get_version(contents: str, version_cls: Type[BaseVersion]) -> BaseVersion:
+    """Parse the version from file contents.
 
     Args:
-        file_ (str): the version file path
-        parent_directory (str): the directory file_ lives under
+        contents (str): The contents of the version.txt file.
+        version_cls (BaseVersion): The `mozilla-version` class to parse the version with.
 
     Returns:
-        str: the version.
-
+        BaseVersion: The parsed version object.
     """
-    abs_path = os.path.join(parent_directory, file_)
-    log.info("Reading {} for version information.".format(abs_path))
-    VersionClass = _find_what_version_parser_to_use(file_, source_repo)
-    with open(abs_path, "r") as f:
-        contents = f.read()
-    log.info("Contents:")
-    for line in contents.splitlines():
-        log.info(" {}".format(line))
     lines = [line for line in contents.splitlines() if line and not line.startswith("#")]
-    return VersionClass.parse(lines[-1])
+    return version_cls.parse(lines[-1])
```

Comment:
Good call on reusing the version parser we already determined earlier in the call stack 👍